File: oracle/jython/create/oracle_create.py
#! /usr/bin/jython
# -*- coding: utf-8 -*-
#
#	oracle_create.py
#
#					Jun/01/2011
#
# --------------------------------------------------------
import	sys
from java.lang import	System
#
import java
from java.sql import DriverManager
#
sys.path.append ('/var/www/data_base/common/jython_common')
sys.path.append ('/var/www/data_base/common/python_common')
from jython_rdb_manipulate import display_proc,drop_table_proc, \
		create_table_proc,table_insert_proc
#
from text_manipulate import dict_append_proc
#
# --------------------------------------------------------
def	data_prepare_proc ():
	dict_aa = {} 
#
	dict_aa = dict_append_proc (dict_aa,'t0131','函館'.decode ('utf-8'),39174,'2004-4-21')
	dict_aa = dict_append_proc (dict_aa,'t0132','札幌'.decode ('utf-8'),17265,'2004-5-12')
	dict_aa = dict_append_proc (dict_aa,'t0133','帯広'.decode ('utf-8'),72654,'2004-6-17')
	dict_aa = dict_append_proc (dict_aa,'t0134','釧路'.decode ('utf-8'),64352,'2004-9-9')
	dict_aa = dict_append_proc (dict_aa,'t0135','旭川'.decode ('utf-8'),42651,'2004-8-4')
	dict_aa = dict_append_proc (dict_aa,'t0136','北見'.decode ('utf-8'),35187,'2004-1-21')
	dict_aa = dict_append_proc (dict_aa,'t0137','室蘭'.decode ('utf-8'),81256,'2004-7-23')
	dict_aa = dict_append_proc (dict_aa,'t0138','根室'.decode ('utf-8'),24784,'2004-10-26')
	dict_aa = dict_append_proc (dict_aa,'t0139','稚内'.decode ('utf-8'),75823,'2004-12-15')
#
	return	dict_aa
#
# --------------------------------------------------------
System.out.println (u"*** 開始 ***")
#
java.lang.Class.forName("oracle.jdbc.driver.OracleDriver")
#
dict_aa = data_prepare_proc ()
#
url="jdbc:oracl:thin:@spn109:1521/xe"
user = "scott"
password = "tiger"
#
conn = DriverManager.getConnection (url,user,password)
#
drop_table_proc (conn)
create_table_proc (conn)
table_insert_proc (conn,dict_aa)
#
display_proc (conn)
#
conn.close ()
System.out.println ("*** 終了 ***")
#
# --------------------------------------------------------

File: oracle/jruby/create/oracle_create.rb
#! /usr/bin/jruby
# -*- encoding: utf-8 -*-
#
#	oracle_create.rb
#
#					Aug/06/2015
# ---------------------------------------------------------------------
include Java
import java.sql.DriverManager
import java.lang.System
import java.util.Locale

import java.util.Date
import java.text.DateFormat
import java.text.SimpleDateFormat
#
#
load '/var/www/data_base/common/ruby_common/text_manipulate.rb'
load '/var/www/data_base/common/jruby_common/jruby_sql_manipulate.rb'
# ---------------------------------------------------------------------
def prepare_data_proc ()
dict_aa={}
dict_aa=dict_append_proc(dict_aa,"t0131","函館",41968,"2007-7-14")
dict_aa=dict_append_proc(dict_aa,"t0132","札幌",27275,"2007-2-27")
dict_aa=dict_append_proc(dict_aa,"t0133","帯広",32824,"2007-11-8")
dict_aa=dict_append_proc(dict_aa,"t0134","釧路",25614,"2007-1-15")
dict_aa=dict_append_proc(dict_aa,"t0135","旭川",84148,"2007-5-21")
dict_aa=dict_append_proc(dict_aa,"t0136","北見",21581,"2007-4-7")
dict_aa=dict_append_proc(dict_aa,"t0137","室蘭",42692,"2007-4-8")
dict_aa=dict_append_proc(dict_aa,"t0138","根室",54125,"2007-8-15")
dict_aa=dict_append_proc(dict_aa,"t0139","稚内",72152,"2007-6-9")
return dict_aa
end
# ---------------------------------------------------------------------
puts	"*** 開始 ***"
#
host = "host_oracle"
user = "scott"
password = "tiger"

str_connect = "jdbc:oracl:thin:@" + host + ":1521/xe"

Java::JavaClass.for_name("oracle.jdbc.driver.OracleDriver")

conn = DriverManager.getConnection(str_connect,user,password)
#
dict_aa=prepare_data_proc()
#
sql_drop_table_proc(conn)
sql_create_table_proc(conn)
#
dict_aa.each {|key,value|
#	puts "key=",key

	str_date = value['date_mod']
	sdf = java.text.SimpleDateFormat("yyyy-MM-dd")
	date_aa = sdf.parse (str_date)
#	date_aa = SimpleDateFormat("yyyy-MM-dd").parse (str_date)
	str_ee = SimpleDateFormat(DATE_PATTERN,Locale.ENGLISH).format(date_aa).new
	
	sql_insert_record_proc(conn,key,value['name'], \
		value['population'],str_ee)
	}
#
sql_read_proc(conn)
#
conn.close()
puts	"*** 終了 ***"
#
# ---------------------------------------------------------------------
